packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart

```diff
@@ -112,6 +143,11 @@ class FirebaseAuthWeb extends FirebaseAuthPlatform {
     return UserWeb(this, _webAuth.currentUser);
   }
 
+  @override
+  set currentUser(UserPlatform userPlatform) {
```

Comment:
Since we've now set `this.currentUser`, the `get currentUser` method should probably use this value, I think. That also means, the event which returns the current user (`onAuthStateChanged`) should also be setting it? 

Right now, the current user is being taken from the `_webAuth` instance, rather than our own `currentUser` property.

Reply:
Updated the currentUser to get our private `_currentUser`. To your second point, if I understood you correctly;   `onAuthStateChanged` is called in a static method on initialisation so we can't set any instance variables. I don't think it should matter because on first use of the delegate, we `setInitialValues` which will set the user and languageCode anyway.

Reply:
Right now though, the `_currentUser` is set by `setInitialValues`, which is fine. However once the web instance is initialised it'll also trigger an [onAuthStateChanged](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart#L68) to pass in those events to the change listeners on the users end.

If the user logs out, the users will get the event through the Stream, but the `currentUser` won't be updated since it now comes from the `_currentUser` property, which hasn't been updated.

Therefore the `_currentUser` needs updating here: https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart#L75